### ISSUE TITLE
Nothing Populating

### ISSUE THREAD

**Issue by sql-chris-albert** [microsoft/AzureStorageExplorer#4768]
### Preflight Checklist

- [X] I have installed the [latest version of Storage Explorer](https://github.com/Microsoft/AzureStorageExplorer/releases/latest).
- [X] I have checked existing resources, including the [troubleshooting guide](https://docs.microsoft.com/azure/storage/common/storage-explorer-troubleshooting) and the [release notes](https://github.com/microsoft/AzureStorageExplorer/releases).
- [X] I have searched for [similar issues](https://github.com/Microsoft/AzureStorageExplorer/issues).

### Storage Explorer Version

1.20.1

### Architecture

x64

### Storage Explorer Build Number

20210720.10

### Platform

Windows

### OS Version

Windows 10 Pro 20H2

### Bug Description

When I open Storage Explorer nothing is populated in either the account view or explorer view. To get my account to show up I have to click View --> Account Management.

### Steps to Reproduce

1. Launch Explorer

### Actual Experience

![image](https://user-images.githubusercontent.com/4333792/128763105-6cba1eaa-0a85-4b9b-b7c1-063b4711ea46.png)

### Expected Experience

I expected my accounts to be there in account view and storage accounts in explorer view.

### Additional Context

This was working on this machine for a long time. Not sure what caused it to break. I have uninstalled\reinstalled several times, removed my account and added back in, and the behavior persists. I have another machine with storage explorer installed and I am able to use it with no issue.

At one point I was using storage explorer in the Azure Portal. I noticed a button that said open in explorer. I clicked that and it launched explorer locally and opened the folder. A message popped up saying the account had been hidden but did not give me any clue as to how that is resolved.

**Comment by MRayermannMSFT** [comment]
@sql-chris-albert hi! Can you do the following for us:
1. While Storage Explorer is open, go to Help -> Toggle Developer tools, and then in the panel/window that opens look for console. Click on console, and then let us know if there's any error/red text
2. [Grab us the Storage Explorer logs.](https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-troubleshooting?tabs=Windows%2C2004#storage-explorer-logs) When doing this, please set that `STG_EX_LOG_LEVEL` environment variable to `debug`. You'll of course need to run storage explorer/reproduce the issue again after setting the environment variable.

**Comment by alexandru-pet** [comment]
Hi Chris, I am facing the same issue, with all the details you described, right after I have upgraded to v1.20.1. I removed the app and downloaded it again, same problem.  By any chance, did you take this up to Microsoft? I would like to upvote there if you did.

**Comment by alexandru-pet** [comment]
> 
> 
> Hi Chris, I am facing the same issue, with all the details you described, right after I have upgraded to v1.20.1. I removed the app and downloaded it again, same problem. By any chance, did you take this up to Microsoft? I would like to upvote there if you did.

FYI - I have uninstalled v1.20.1 and installed v.1.19.1 and it works fine.

**Comment by sql-chris-albert** [comment]
@alexandru-pet Thanks for the tip. I installed 1.19.1 and everything is now working. Hopefully MS will be able to resolve the issue with the new version.

**Comment by MRayermannMSFT** [comment]
@alexandru-pet @sql-chris-albert 

Albert's logs did not reveal anything to us about what is potentially going wrong. I've created a private build with additional logging. If y'all could download it, run it, [gather the storage explorer logs](https://docs.microsoft.com/en-us/azure/storage/common/storage-explorer-troubleshooting?tabs=Windows%2C2004#storage-explorer-logs) (make sure to set that logs environment variable) and share the logs with us that'd be a big help.

I have two options for running the build:

- Standalone zip: https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-1/StorageExplorer-win32-ia32.zip
- Normal installer: https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-1/StorageExplorer-ia32.exe

Since y'all have downgraded to 1.19.1 for now, then you can use the zip to run the private build without losing your current install. Just download the zip, unzip, and then run the .exe in the root folder.

Thanks!

**Comment by sql-chris-albert** [comment]
@MRayermannMSFT I downloaded the Zip and unzipped. When I execute the .exe it show a blank window for a second and then nothing. Have tried a few times.

**Comment by MRayermannMSFT** [comment]
@sql-chris-albert that's weird....I guess if you have time then to do the full install that would be appreciated. Unfortunately you will have to reinstall 1.19.1 afterwards. ☹

**Comment by sql-chris-albert** [comment]
[2021-08-17_152227_22792.log](https://github.com/microsoft/AzureStorageExplorer/files/7002501/2021-08-17_152227_22792.log)
[2021-08-17_152227_main.log](https://github.com/microsoft/AzureStorageExplorer/files/7002502/2021-08-17_152227_main.log)
[2021-08-17_152227_renderer.log](https://github.com/microsoft/AzureStorageExplorer/files/7002504/2021-08-17_152227_renderer.log)
[2021-08-17_152227_se-markdown-panel_9444.log](https://github.com/microsoft/AzureStorageExplorer/files/7002505/2021-08-17_152227_se-markdown-panel_9444.log)

**Comment by MRayermannMSFT** [comment]
Awesome thanks! I can already see that something is "off" compared to what I see on my machine. I'll digest this and get back to you. Thank you thank you thank you a million times. :)

**Comment by MRayermannMSFT** [comment]
Ok I think those logs have helped me narrow down the area the problem is occurring in. I have a new private build with yet more logging:

- Standalone zip: https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-2/StorageExplorer-win32-ia32.zip
- Normal installer: https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-2/StorageExplorer-ia32.exe

**Comment by sql-chris-albert** [comment]
[2021-08-18_091855_996.log](https://github.com/microsoft/AzureStorageExplorer/files/7007550/2021-08-18_091855_996.log)
[2021-08-18_091855_main.log](https://github.com/microsoft/AzureStorageExplorer/files/7007551/2021-08-18_091855_main.log)
[2021-08-18_091855_renderer.log](https://github.com/microsoft/AzureStorageExplorer/files/7007552/2021-08-18_091855_renderer.log)
[2021-08-18_091855_se-markdown-panel_23292.log](https://github.com/microsoft/AzureStorageExplorer/files/7007553/2021-08-18_091855_se-markdown-panel_23292.log)

**Comment by MRayermannMSFT** [comment]
Sorry it took me a while to get back to you @sql-chris-albert ...

So the latest round of logs give more insight into where Storage Explorer is not working right but I still do not have a lot of insight into **why**. And on top of that, the **where** that I have identified is one of the more opaque/black-box parts of our product.

So with all of that in mind I'd like to try something different today. This will be a bit more involved on your part, but if you have the time for it that would be helpful.

I've prepared a couple of private builds. These builds are from commits (aka changes) we made while developing version 1.20. I'd like to use these private builds to help us narrow down when the breaking change was made to Storage Explorer. I've looked through our commits for 1.20, and these ones are my current best guesses for what could be inadvertently causing this problem. What you need to do is simply install the build and find the oldest one with which this issue reproduces on your machine. I do not need you to get me any log files.

So here are the builds:

- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds/StorageExplorer-ia32-c6194393.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds/StorageExplorer-ia32-c4a2872b.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds/StorageExplorer-ia32-715c7ea5.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds/StorageExplorer-ia32-c5d38b93.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds/StorageExplorer-ia32-d2121ada.exe

They are ordered oldest to newest. I suggest you trying them out in either of the following orders:
- Try the middle one first (715c7ea5), and if the issue does repro, then work your way backwards from there, or if it doesn't repro, then work your way forwards from there.
- Oldest to newest.

Let me know if you have any questions. 

Also, @alexandru-pet if you are still around feel free to partake in this as well.

**Comment by sql-chris-albert** [comment]
I started in the middle with 715c7ea5, and it worked fine. c5d38b93 worked fine as well. d2121ada is where the issue came back up.

**Comment by MRayermannMSFT** [comment]
@sql-chris-albert super! I will delve deeper into our changes from that time period!

**Comment by MRayermannMSFT** [comment]
Hey @sql-chris-albert (and @alexandru-pet), I have a new set of builds to try. Apologies to ask you to keep doing this, but since we cannot repro the issue and logging did not reveal the issue this seems to be the best course of action.

Right now I feel as if `d2121ada` is what caused the issue. Out of all the possible changes from `c5d38b93` to `d2121ada` (inclusive) it is the one that comes the closest to touching what is failing to finish. So to verify that, I have a build of the commit before `d2121ada`, along with a couple of others that occurred  after `c5d38b93` but before `d2121ada` that are also suspect. Here they are:

- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds-2/StorageExplorer-ia32-bcce041d.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds-2/StorageExplorer-ia32-b0eed2b6.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds-2/StorageExplorer-ia32-e08eb44f.exe
- https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds-2/StorageExplorer-ia32-2c2efa25.exe

The `2c2efa25` exe is the commit directly before `d2121ada`, so I guess I'd recommend:

1. Try `2c2efa25`, if issue doesn't repro then you're done/we've found the breaking change
2. If issues does repro then try `b0eed2b6`
3. If issue does repro then try  `bcce041d` or if it doesn't repro then try `e08eb44f`

Also, if you're curious, we're working on a range of changes from May 12th to June 22nd. So there's a lot of commits to sift through. I'm glad that we were able to so greatly reduce it last night though! The original range of changes was April 15th to June 22nd. 😬

Finally, if my guess about `d2121ada` is right then we'll have to do some experiments to figure out why the heck that change is breaking the tree view on your machine. The change was just the removing of sending telemetry in an error handler which makes me feel like 🤷‍♂️.

**Comment by alexandru-pet** [comment]
Hi guys, sorry for not being able to join you on this investigation so far. I had some work related tasks for which I was highly dependant on the logs and couldn't risk losing access to them even for an hour. Today I am off early, but I will try to join from Monday.

**Comment by MRayermannMSFT** [comment]
Cool thanks for finding the time @alexandru-pet!

**Comment by sql-chris-albert** [comment]
1. Tried 2c2efa25 and it did not work.
2. Tried b0eed2b6 and it did not work.
3. Tried bcce041d and it did not work.

**Comment by MRayermannMSFT** [comment]
Ok  @sql-chris-albert & @alexandru-pet, one more build to try: 

https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-dev-builds/StorageExplorer-ia32-41178ff2.exe

With regards to Chris' findings:
- So my guess about `d2121ada` being the cause was wrong.
- Commit `bcce041d` happened shortly after `c5d38b93`, and this latest build is right before `bcce041d`. 

Some additional questions (I have the answers to these from the log files but I want to cross check them with y'all):
- You have **not** enabled proxy settings in Storage Explorer. Correct?
- Storage Explorer is running in **English**. Correct?

**Comment by sql-chris-albert** [comment]
41178ff2 worked.

I have not enabled proxy settings and my language is set to use system settings which is English.

**Comment by MRayermannMSFT** [comment]
Ok, I have a new build to try, and now we're back to sharing logs! I've decided to ignore the I can't fathom how that change broke the tree view init'ing and have added logging in the area of the change and also in additional areas that I didn't have before.

@sql-chris-albert & @alexandru-pet, when y'all have some free time please install, generate logs, gather logs, and then share logs from this build:

https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-3/StorageExplorer-ia32.exe

Thanks.

**Comment by sql-chris-albert** [comment]
[2021-08-24_095030_20156.log](https://github.com/microsoft/AzureStorageExplorer/files/7039878/2021-08-24_095030_20156.log)
[2021-08-24_095030_main.log](https://github.com/microsoft/AzureStorageExplorer/files/7039879/2021-08-24_095030_main.log)
[2021-08-24_095030_renderer.log](https://github.com/microsoft/AzureStorageExplorer/files/7039881/2021-08-24_095030_renderer.log)
[2021-08-24_095030_se-markdown-panel_22356.log](https://github.com/microsoft/AzureStorageExplorer/files/7039882/2021-08-24_095030_se-markdown-panel_22356.log)

**Comment by MRayermannMSFT** [comment]
@sql-chris-albert & @alexandru-pet

Ok. Still eluding us. I've gone crazy on today's build and have some additional things to gather/ask. First here's the build: https://storageexplorerpublish.blob.core.windows.net/privatebuilds/4768-4/StorageExplorer-ia32.exe

So, for this build:
- When you run it, let it sit for at least 30 seconds after the main window appears. I've added some "retry/give up waiting for things to say they are ready" logic to try and generate some actual errors (so far we've seen no errors in any of the logging we've added 😢).
- After letting it sit:
    - As normal, gather the log files, and then share them here
    - Open the dev tools with F12/via the help menu, go to the console tab, and filter by `CloudExplorer.App.` and then share what you find. Here's an example of what I mean:
![image](https://user-images.githubusercontent.com/21112697/130707031-bbb4e1c9-41ae-4e84-93ef-fcd914baac19.png)

Also some additional questions:
- Are you reproducing this on any other machines?
- For all machines you are reproducing on, in what ways are those machines similar? For example:
    - Are they all managed by your companies IT?
    - Are they running any anti virus software other than Windows Defender?
    - Do you have any problems with other applications on these machines? For example: Microsoft Teams, VS Code, Slack, Discord, Postman, GitKraken?

**Comment by sql-chris-albert** [comment]
I'm out of the office tomorrow. Will get this done on Thursday.

**Comment by sql-chris-albert** [comment]
@MRayermannMSFT Any luck with the last round of logs?

**Comment by MRayermannMSFT** [comment]
@sql-chris-albert sorry, been busy getting ready to ship 1.21.0. No, the new logs have not been anymore helpful. The tree view is proving to be a black hole from which no information escapes...

- Is the Windows user you are using the same one that installed Storage Explorer?
- Can you try running as admin?
- Can you try changing theme after the app has launched?
- Can you try using the default or another theme?
- Can you go into the F12 tools, go to Elements `ctrl+f` for `cloud-explorer-panel`, find the first result and then...
   1. Take a screenshot of that first result (similar to mine below, but leave out the context menu)
   2. Right click on the `<html` line, choose "Copy element", save that to a text file, and then attach that file here (see screenshot to see what I'm talking about)
![image](https://user-images.githubusercontent.com/21112697/132424370-dc10b08c-6ea2-48a7-87de-45974e834174.png)
- Attach this file here as well: `C:/Users/calbert/AppData/Local/Programs/Microsoft Azure Storage Explorer/resources/app/out/CloudHub/html/CloudExplorer/index.html`, make sure your user account is able to open and read this file as well.

**Comment by JasonYeMSFT** [comment]
This issue might be caused by us loading modules take longer than expected and timeout. For all of you who run into this problem, could you please try modifying a source file to override the default timeout parameter and see if mitigates the problem?

The file to modify:

- Windows: 
- macOS: `/Applications/Microsoft Azure Storage Explorer.app/Contents/Resources/app/out/CloudHub/js/CloudExplorer/config.js`

Somewhere at the end of config.js, add a comma and add one line below. Basically we are going to add a `waitSeconds` parameters to the `require` object on `window`. This would extend the timeout period to 30 seconds instead of the default 7 seconds.

Before:
```js
window.require = {
    baseUrl: "../../",
    paths: {
        "ccs": "internal/ccs/ccs-0.1.0",
        "es6-promise": "lib/es6-promise/es6-promise",
        "jquery": "lib/jquery/dist/jquery",
        "knockout": "lib/knockout/dist/knockout",
        "knockout.mapping": "lib/knockout-mapping/knockout.mapping",
        "underscore": "lib/underscore/underscore",
        "underscore.string": "lib/underscore.string/dist/underscore.string",
        "URIjs": "lib/uri.js/src",
        "ActivityLog": "js/ActivityLog",
        "CloudExplorer": "js/CloudExplorer",
        "Common": "js/Common",
        "Providers": "js/Providers",
        "StorageExplorer": "js/StorageExplorer",
        "text": "lib/text/text"
    },
    shim: {
        jquery: {
            exports: "$"
        },
        knockout: {
            deps: ["jquery"],
            exports: "ko"
        },
        "knockout.mapping": {
            deps: ["jquery", "knockout"],
            exports: "mapping"
        }
    }
};
```

After:
```js
window.require = {
    baseUrl: "../../",
    paths: {
        "ccs": "internal/ccs/ccs-0.1.0",
        "es6-promise": "lib/es6-promise/es6-promise",
        "jquery": "lib/jquery/dist/jquery",
        "knockout": "lib/knockout/dist/knockout",
        "knockout.mapping": "lib/knockout-mapping/knockout.mapping",
        "underscore": "lib/underscore/underscore",
        "underscore.string": "lib/underscore.string/dist/underscore.string",
        "URIjs": "lib/uri.js/src",
        "ActivityLog": "js/ActivityLog",
        "CloudExplorer": "js/CloudExplorer",
        "Common": "js/Common",
        "Providers": "js/Providers",
        "StorageExplorer": "js/StorageExplorer",
        "text": "lib/text/text"
    },
    shim: {
        jquery: {
            exports: "$"
        },
        knockout: {
            deps: ["jquery"],
            exports: "ko"
        },
        "knockout.mapping": {
            deps: ["jquery", "knockout"],
            exports: "mapping"
        }
    },
    waitSeconds: 30 // line added
};
```

**Comment by sql-chris-albert** [comment]
@JasonYeMSFT That worked. Thanks so much!

**Comment by JasonYeMSFT** [comment]
Nice. We will evaluate the timeout parameter in our next milestone and extend it as needed.

**Comment by JasonYeMSFT** [comment]
We have shipped a patched version 1.22.1 which bumps all related timeout parameters to 30 seconds.
